tests/path_gateway_raw_test.go

```diff
@@ -157,3 +143,117 @@ func TestGatewayBlock(t *testing.T) {
+							return v != ""
+						}),
+					Header("Content-Range").
+						ChecksAll(func(v []string) bool {
+							if len(v) == 1 {
+								contentRange = v[0]
+							}
+							return true
+						}),
+				),
+		},
+	}, specs.PathGatewayRaw)
+
+	if contentType == "application/vnd.ipld.raw" {
+		// The server is not able to respond to a multi-range request. Therefore,
+		// there might be only one range or... just the whole file, depending on the headers.
+
+		if contentRange == "" {
+			// Server does not support range requests and must send back the complete file.
+			RunWithSpecs(t, SugarTests{
+				{
+					Name: "GET with application/vnd.ipld.raw with multiple range request includes correct bytes",
+					Request: Request().
+						Path("/ipfs/{{cid}}/dir/ascii.txt", fixture.MustGetCid()).
+						Headers(
+							Header("Accept", "application/vnd.ipld.raw"),
+							Header("Range", "bytes=6-16,0-4"),
+						),
+					Response: Expect().
+						Status(206).
+						Headers(
+							Header("Content-Type", "application/vnd.ipld.raw"),
+							Header("Content-Range").IsEmpty(),
+						).
+						Body(fixture.MustGetRawData("dir", "ascii.txt")),
+				},
+			}, specs.PathGatewayRaw)
+		} else {
+			// Server supports range requests but only the first range.
+			RunWithSpecs(t, SugarTests{
+				{
+					Name: "GET with application/vnd.ipld.raw with multiple range request includes correct bytes",
+					Request: Request().
+						Path("/ipfs/{{cid}}/dir/ascii.txt", fixture.MustGetCid()).
+						Headers(
+							Header("Accept", "application/vnd.ipld.raw"),
+							Header("Range", "bytes=6-16,0-4"),
+						),
+					Response: Expect().
+						Status(206).
+						Headers(
+							Header("Content-Type", "application/vnd.ipld.raw"),
+							Header("Content-Range", "bytes 6-16/31"),
+						).
+						Body(fixture.MustGetRawData("dir", "ascii.txt")[6:17]),
+				},
+			}, specs.PathGatewayRaw)
+		}
+	} else if strings.Contains(contentType, "multipart/byteranges") {
+		// The server supports responding with multi-range requests.
+		RunWithSpecs(t, SugarTests{
+			{
+				Name: "GET with application/vnd.ipld.raw with multiple range request includes correct bytes",
+				Request: Request().
+					Path("/ipfs/{{cid}}/dir/ascii.txt", fixture.MustGetCid()).
+					Headers(
+						Header("Accept", "application/vnd.ipld.raw"),
+						Header("Range", "bytes=6-16,0-4"),
+					),
+				Response: Expect().
+					Status(206).
+					Headers(
+						Header("Content-Type").Contains("multipart/byteranges"),
+					).
+					Body(And(
+						Contains("Content-Range: bytes 6-16/31"),
+						Contains("Content-Type: application/vnd.ipld.raw"),
+						Contains(string(fixture.MustGetRawData("dir", "ascii.txt")[6:17])),
+						Contains("Content-Range: bytes 0-4/31"),
+						Contains(string(fixture.MustGetRawData("dir", "ascii.txt")[0:5])),
+					)),
+			},
+		}, specs.PathGatewayRaw)
+	} else {
+		t.Error("Content-Type header did not match any of the accepted options")
+	}
+}
```

Comment:
One piece of advice here:
Create a test array variable, and append to it based on your if / else, then at the end call `RunWithSpecs(t, yourTests)`

Similar to what we do in [other test](https://github.com/ipfs/gateway-conformance/blob/182dd476f207776e3a95598b6b4f81d02fae7ab5/tests/redirects_file_test.go#L41),

It'll be clearer for future maintainainers what is the content of the test (the if / else) and what is part the test framework (the run...)

Reply:
Ok, I did that. Do you think I should keep the error as it is, or should that be done somewhere else?

Reply:
We could use a `t.FailNow()` to exit early, but I have no strong opinion here.